src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,231 @@
+        try {
+            Integer idx = Integer.parseInt(parser.getDescription(input));
+            Task item = listOfItems.getTask(idx);
+            listOfItems.removeTask(idx);
+            ui.printDeletionConfirmation(item, listOfItems);
+        } catch (NumberFormatException e) {
+            throw new DukeException("you need to choose a number from the list in the form: delete (list index)!");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("you need to pick an item from within the list!");
+        }
+    }
+
+    /**
+     * Displays a list of items on the command line that contain the keyword entered by the user.
+     *
+     * @param input Command line input from the user.
+     * @throws DukeException Throws an exception if no keyword was entered or if no tasks match the keyword given.
+     */
+    private void findItems(String input) throws DukeException {
+        try {
+            String keyword = parser.getDescription(input);
+            Tasklist validItems = listOfItems.findAllBy(keyword);
+            ui.outputWrapper(validItems);
+        }catch (ArrayIndexOutOfBoundsException e){
+            throw new DukeException("please input a keyword in the format: [find] (keyword)");
+        } catch (NullPointerException e) {
+            throw new DukeException("no tasks match your given keyword!");
+        }
+
+    }
+    /**
+     * Handles invalid commands from the user.
+     *
+     * @throws DukeException Throws an exception when a wrong input is given on the command line.
+     */
+    private void markAsInvalid(String input) throws DukeException {
+        if (input.equals("")) {
+            throw new DukeException("you need to type something!");
+        } else {
+            throw new DukeException("your command is invalid!");
+        }
+    }
+
+    /**
+     * Handles the user's input and determines which command should be run.
+     *
+     * @param input Command line input from the user.
+     */
+    private void handleInput(String input) {
+        String command = parser.getCommand(input);
+        try {
+            switch(command) {
+                case "items":
+                    ui.outputWrapper(listOfItems);
+                    break;
+                case "completed":
+                    markAsDone(input);
+                    break;
+                case "todo":
+                    addAsTodo(input);
+                    break;
+                case "event":
+                    addAsEvent(input);
+                    break;
+                case "deadline":
+                    addAsDeadline(input);
+                    break;
+                case "delete":
+                    deleteItem(input);
+                    break;
+                case "find":
+                    findItems(input);
+                    break;
+                default:
+                    markAsInvalid(input);
+                    break;
```

Comment:
should the indentation reduce a tap space?

Reply:
yep thanks i missed this part out